Modules/SubModule_Import/xASL_imp_NII2BIDS_Run.m

```diff
@@ -1,59 +1,89 @@
-        xASL_imp_NII2BIDS_RunIssueWarning(loggingEntry, 'anatomical', subjectSessionLabel, iRun);
-    end
-    
-    %% Perfusion files
-    try
-        xASL_imp_NII2BIDS_RunPerf(imPar, bidsPar, studyPar, subjectSessionLabel, inSessionPath, outSessionPath, listRuns, iRun);
-    catch loggingEntry
-        [x] = xASL_qc_AddLoggingInfo(x, loggingEntry);
-        xASL_imp_NII2BIDS_RunIssueWarning(loggingEntry, 'perfusion', subjectSessionLabel, iRun);
+    %% 1. Make a subject directory with a correct session name
+    if ~isempty(bidsLabel.visit)
+        %if ~isempty(bidsLabel.visit)
+        sessionLabel = ['ses-' bidsLabel.visit];
+        %else
+        %    sessionLabel = ['ses-' listRuns{iRun}(5:end)];
+        %end
+        
+        % Create directory if it does not exist already
+        sessionPerfusionDirectory = fullfile(imPar.BidsRoot,['sub-' bidsLabel.subject], sessionLabel,'perf');
+        xASL_adm_CreateDir(sessionPerfusionDirectory);
+        
+        inSessionPath = fullfile(imPar.TempRoot, nameSubjectSession, listRuns{iRun});
+        outSessionPath = fullfile(imPar.BidsRoot, ['sub-' bidsLabel.subject], sessionLabel);
+
+        % Need to add the underscore so that it doesn't need to be added automatically and can be skipped for empty session
+        sessionLabel = ['_' sessionLabel];
+    else
+        % Session label is skipped
+        sessionLabel = '';
+
+        % Only one session - no session labeling
+        if ~exist(fullfile(imPar.BidsRoot,['sub-' bidsLabel.subject]),'dir')
+            xASL_adm_CreateDir(fullfile(imPar.BidsRoot, ['sub-' bidsLabel.subject], bidsPar.strPerfusion));
+        end
+        inSessionPath = fullfile(imPar.TempRoot, nameSubjectSession, listRuns{iRun});
+        outSessionPath = fullfile(imPar.BidsRoot, ['sub-' bidsLabel.subject]);
     end
-    
-end
 
 
+    %% 2. Convert structural and ASL runs
+	subjectSessionLabel = [bidsLabel.subject sessionLabel];
+	try
+		xASL_imp_NII2BIDS_RunAnat(imPar, bidsPar, studyPar, subjectSessionLabel, outSessionPath, listRuns, iRun, nameSubjectSession);
+	catch loggingEntry
+		[x] = xASL_qc_AddLoggingInfo(x, loggingEntry);
+		xASL_imp_NII2BIDS_RunIssueWarning(x,loggingEntry, 'anatomical', subjectSessionLabel, iRun);
+	end
+		
+	%% Perfusion files
+	try
+		xASL_imp_NII2BIDS_RunPerf(imPar, bidsPar, studyPar, subjectSessionLabel, inSessionPath, outSessionPath, listRuns, iRun);
+	catch loggingEntry
+		[x] = xASL_qc_AddLoggingInfo(x, loggingEntry);
+		xASL_imp_NII2BIDS_RunIssueWarning(x,loggingEntry, 'perfusion', subjectSessionLabel, iRun);
+	end
+
+end
+
 %% Issue failed run as a warning
-function xASL_imp_NII2BIDS_RunIssueWarning(loggingEntry, Scantype, subjectSessionLabel, iRun)
+function xASL_imp_NII2BIDS_RunIssueWarning(x,loggingEntry, Scantype, subjectSessionLabel, iRun)
 
-    fprintf('\n==============================================================================================\n');
+    fprintf(x.design.breakString);
```

Comment:
Will we use this also in the other modules, or only here? So should it be `x.modules.import.design` instead or not?

Reply:
Let's just keep it where it is for now. Don't worry, I was cleaning the x struct a lot in the past, so it should be okay to add a single new field.